lib/import-export-visitor.js

```diff
@@ -303,15 +318,16 @@ IEVp.finalizeHoisting = function () {
     delete bodyInfo.hoistedExportsMap;
     delete bodyInfo.hoistedExportsString;
     delete bodyInfo.hoistedImportsString;
-  }, this);
+  }
 
   // Just in case we call finalizeHoisting again, don't hoist anything.
   this.bodyInfos.length = 0;
 
-  this.removals.forEach(processRemoval);
+  let removalCount = this.removals.length;
 
-  // Just in case we call finalizeHoisting again, don't remove anything.
-  this.removals.length = 0;
+  while (removalCount--) {
+    processRemoval(this.removals.pop());
```

Comment:
I was wondering if order mattered. It didn't fail tests. If it's removing things could it have a meaningful change?
Can you give an example? In regards to `pop`, arrays :heart: `pop`, setting the `.length` isn't ness a happy path for engines.